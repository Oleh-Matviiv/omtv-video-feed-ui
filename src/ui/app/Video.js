import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Container = styled.article.attrs({
  className: 'video',
})`
  border: .1rem solid #efefef;
  margin-bottom: 2rem;
  display: grid;
  grid-template-areas: "title views"
                       "play play";
`;
export const Title = styled.div.attrs({
  className: 'video-title',
})`
  padding: 1rem;
  grid-area: "title";
`;
export const Views = styled.div.attrs({
  className: 'video-views',
})`
  padding: 1rem;
  grid-area: "views";
  text-align: right;
`;
export const Play = styled.div.attrs({
  className: 'video-play',
})`
  grid-area: "play";
`;

const Video = ({
  title = '',
  type = 'video',
  source,
  video = 'video unavailable',
  views = 'no info',
}) => (
  <Container>
    <Title>{ title }</Title>
    <Views>{ views }</Views>
    <Play>{ video }</Play>
  </Container>
);

Video.propTypes = {
  title: PropTypes.string,
  type: PropTypes.oneOf(['video']),
  source: PropTypes.oneOf([
    'url', 'youtube', 'facebook',
  ]).isRequired,
  video: PropTypes.string,
  views: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

export default Video;
