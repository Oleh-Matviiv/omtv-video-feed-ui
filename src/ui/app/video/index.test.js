import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Video from './';

describe('snapshots', () => {
  it('Video url', () => {
    const wrapper = shallow(
      <Video
        title="url based video"
        source="url"
        video="http://g.com/1.mp4"
        views={3445}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Video youtube', () => {
    const wrapper = shallow(
      <Video
        title="youtube based video"
        source="youtube"
        video="Y34fdvsf43"
        views={214526}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Video facebook', () => {
    const wrapper = shallow(
      <Video
        title="facebook based video"
        source="facebook"
        video="F34fdvsf43"
        views={56789}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
