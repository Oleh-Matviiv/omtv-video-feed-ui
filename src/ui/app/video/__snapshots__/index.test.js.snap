// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Video facebook 1`] = `
<styled.article>
  <styled.div>
    facebook based video
  </styled.div>
  <styled.div>
    56789
  </styled.div>
</styled.article>
`;

exports[`snapshots Video url 1`] = `
<styled.article>
  <styled.div>
    url based video
  </styled.div>
  <styled.div>
    3445
  </styled.div>
  <Url
    src="http://g.com/1.mp4"
  />
</styled.article>
`;

exports[`snapshots Video youtube 1`] = `
<styled.article>
  <styled.div>
    youtube based video
  </styled.div>
  <styled.div>
    214526
  </styled.div>
</styled.article>
`;
